src/please.go

```diff
@@ -453,7 +453,8 @@ var buildFunctions = map[string]func() bool{
 					config.Cache.RPCURL = ""
 					config.Cache.HTTPURL = ""
 				}
-				clean.Clean(config, newCache(config), !opts.Clean.NoBackground)
+				state := core.NewBuildState(1, nil, 4, config)
```

Comment:
This feels a bit odd too - what are these magic numbers?

Reply:
Number of threads & verbosity (can't actually remember why that's in there tbh, it feels like it shouldn't be).

Agreed it is a mess - I want to clean up those constructors at some point, they aren't really super useful as they are.